### DISCUSSION TITLE
Know which key was used

### DISCUSSION THREAD

**Discussion by kboirel** [krisk/Fuse#589] [Q&A] [answered]
I use Fuse to filter a database. I want to search among names and translations.other_language 

```
const options = {
         threshold: 0.3,
         keys: ['name', 'translations.FR']
}
```

So if I type Apple I can have an object with Apple as a name, but if I type "Pomme" I can also have the same object (thanks to the French translation).

Now I would like to know if the item found, has been found the "name" or with "translations.FR".
Is it possible ?

**Comment by krisk** [comment]
Difficult to help here without seeing what a sample data object looks like, and possibly what you've tried to do.

**Comment by krisk** [comment, marked as answer]
Adding a better answer to my earlier reply — yes, this is possible. Set `includeMatches: true` in your options, and each result will include a `matches` array that tells you which key was matched:

```js
const fuse = new Fuse(list, {
  threshold: 0.3,
  keys: ['name', 'translations.FR'],
  includeMatches: true
})

const results = fuse.search('Pomme')
// results[0].matches[0].key === 'translations.FR'
```

Each entry in `matches` has a `key` property indicating which field produced the match.
